build/generate-style-code.js

```diff
@@ -44,8 +44,8 @@ global.propertyType = function (property) {
         return `DataDrivenProperty<${flowType(property)}>`;
     } else if (/-pattern$/.test(property.name) || property.name === 'line-dasharray') {
         return `CrossFadedProperty<${flowType(property)}>`;
-    } else if (property.name === 'heatmap-color') {
-        return `HeatmapColorProperty`;
+    } else if (property.name === 'heatmap-color' || property.name === 'line-gradient') {
```

Comment:
It's great that we have a second property with the same semantics as `heatmap-color` -- that increases the likelihood that `heatmap-color` is not some odd exception (and as we know, the odd exceptions are disproportionately expensive to maintain).

Since we now have two similarly-behaving properties, it's worth finding a generic way to capture that the style spec and replace the property name comparisons here with the use of a semantic property like `isDataDriven` uses. Such a property would probably be useful for Studio as well. As a schema design question, it intersects with #4194 -- although in light of this new information, I'm not sure my proposal there is still the best choice. @lbud do you want to think about this and come up with a proposal?